sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java

```diff
@@ -167,6 +171,17 @@ public String toString() {
      * from the current transform or component.
      */
     ItemBuilder add(String key, Class<?> value);
+
+    /**
+     * Register the given display metadata. The input value will be inspected to see if it conforms
+     * to one of the supported DisplayData types. Otherwise, it will be registered as a
+     * {@link DisplayData.Type#STRING}, using the {@link Object#toString()} method to retrieve the
+     * display value.
+     *
+     * <p> The added display data is identified by the specified key and namespace from the current
+     * transform or component.
+     */
+    ItemBuilder add(String key, Object value);
```

Comment:
This bugs me, it's too easy to do the wrong thing. I imagine users doing something like `builder.add(mySubcomponent)`, expecting it to be recursively included rather than using `.toString()`.

I'll think about this more...